### PR TITLE
Remove OTA demo network manager files

### PR DESCRIPTION
**Pull request by divekarshubham**
Remove OTA demo network manager files

Description
-----------

* These files are used to switch between different networks for the OTAv1 demo. BLE support will be added with changes to transport interface in a further PR.
* Updated Project files and CMake to remove the reference.

Checklist:
----------

- [ ] I have tested my changes. No regression in existing tests.
- [ ] My code is Linted.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: projects/pc/windows/visual_studio/aws_demos/aws_demos.vcxproj.filters [hunk 1]
```diff
@@ -1234,6 +1232,9 @@
 		<ClCompile Include="..\..\..\..\..\demos\jobs_for_aws\jobs_demo.c">
 			<Filter>demos\jobs_for_aws</Filter>
 		</ClCompile>
+		<ClCompile Include="..\..\..\..\..\demos\ota\ota_demo_core_http\ota_demo_core_http.c">
```
Comment (ravibhagavandas) [inline projects/pc/windows/visual_studio/aws_demos/aws_demos.vcxproj.filters:1235]: Is this file added intentionally?

File: demos/CMakeLists.txt [hunk 2]
```diff
@@ -9,7 +9,6 @@ afr_module_sources(
         "$<$<TARGET_EXISTS:AFR::freertos_plus_tcp>:${AFR_DEMOS_DIR}/demo_runner/aws_demo_network_addr.c>"
         "${AFR_DEMOS_DIR}/demo_runner/aws_demo.c"
         "${AFR_DEMOS_DIR}/network_manager/aws_iot_network_manager.c"
-        "${AFR_DEMOS_DIR}/network_manager/aws_iot_demo_network.c"
```
Comment (ravibhagavandas) [inline demos/CMakeLists.txt:12]: Could you also remove header file `demos/include/aws_iot_demo_network.h`

File: projects/pc/windows/visual_studio/aws_demos/aws_demos.vcxproj.filters (hunk 1 above)
Comment (divekarshubham) [reply projects/pc/windows/visual_studio/aws_demos/aws_demos.vcxproj.filters:1235]: Yes, the project file were not updated after addition on OTA-HTTP demos

File: demos/CMakeLists.txt (hunk 2 above)
Comment (divekarshubham) [reply demos/CMakeLists.txt:12]: Thanks for pointing it out, removed.

**Review by ravibhagavandas — DISMISSED** [review]

**Review by ravibhagavandas — APPROVED** [review]

**Review by aggarg — APPROVED** [review]

**Pull request merged**